src/plugins/intel_cpu/src/emitters/tpp/common/kernel_executors/brgemm.hpp

```diff
@@ -0,0 +1,126 @@
+        return !(*this == rhs);
+    }
+
+    void update(int64_t M, int64_t N, int64_t K, int64_t LDA, int64_t LDB, int64_t LDC, float beta) override;
+
+    size_t hash() const override {
+        return m_hash;
+    }
+    size_t compute_hash() const;
+
+    libxsmm_bitfield get_compile_flags() const {
+        return m_compile_flags;
+    }
+    void set_compile_flags_with_zero_beta(bool zero_beta) {
+        if (zero_beta) {
+            m_compile_flags = m_static_params->m_compile_flags | LIBXSMM_GEMM_FLAG_BETA_0;
+        } else {
+            m_compile_flags = m_static_params->m_compile_flags;
+        }
+    }
+    bool get_prefetching_flags() const {
+        return m_static_params->m_prefetching_flags;
+    }
+    libxsmm_datatype get_type_in0() const {
+        return m_static_params->m_type_in0;
+    }
+    libxsmm_datatype get_type_in1() const {
+        return m_static_params->m_type_in1;
+    }
+    libxsmm_datatype get_type_out0() const {
+        return m_static_params->m_type_out0;
+    }
+    libxsmm_datatype get_type_exec() const {
+        return m_static_params->m_type_exec;
+    }
+#ifdef SNIPPETS_DEBUG_CAPS
+    std::string to_string() const override;
+#endif
+
+private:
+    struct StaticParams {
+        StaticParams(const element::Type& in0_dtype, const element::Type& in1_dtype);
+        virtual ~StaticParams() = default;
+
+        bool operator==(const StaticParams& rhs) const;
+        bool operator!=(const StaticParams& rhs) const {
+            return !(*this == rhs);
+        }
+        size_t hash() const {
+            return m_hash;
+        }
+        size_t compute_hash();
+
+#ifdef SNIPPETS_DEBUG_CAPS
+        std::string to_string() const;
+#endif
+
+        libxsmm_datatype m_type_in0;
+        libxsmm_datatype m_type_in1;
+        libxsmm_datatype m_type_out0;
+        libxsmm_datatype m_type_exec;
+        libxsmm_bitfield m_compile_flags;
+        bool m_prefetching_flags;
+
+        size_t m_hash{SIZE_MAX};
+    };
+    const std::shared_ptr<StaticParams>& get_static_params() const {
+        return m_static_params;
+    }
+
+    libxsmm_bitfield m_compile_flags{0};
+    std::shared_ptr<StaticParams> m_static_params{nullptr};
+
+    size_t m_hash{SIZE_MAX};
+};
+
+// The `update_kernel` method verifies that a compiled kernel is not nullptr.
+// However, the compiled kernel might be empty in cases if nothing is to be compiled (`Config.is_empty() == true`).
+// To cover this case, we wrap the `libxsmm_gemmfunction` in the separate structure which may contain empty
+// `libxsmm_gemmfunction`
+struct BrgemmTppCompiledKernel {
+    std::shared_ptr<libxsmm_gemmfunction> brgemm_kernel = nullptr;
+};
+
+class BrgemmKernelExecutor : public CPUKernelExecutor<BrgemmKernelConfig, BrgemmTppCompiledKernel> {
+public:
+    BrgemmKernelExecutor(ov::intel_cpu::MultiCacheWeakPtr kernel_cache, BrgemmKernelConfig config);
+    virtual ~BrgemmKernelExecutor() = default;
```

Comment:
Minor: it looks like virtual destructor is not necessary: parent class already has it